### ISSUE TITLE
Caliblur has squished covers - fix inside

### ISSUE THREAD

**Issue by lyonzy** [janeczku/calibre-web#1314]
**Describe the bug/problem**
The CaliBlur theme (which looks great by the way!) squishes the covers of some/most books, e.g.:
![book cover with changed aspect ratio](https://user-images.githubusercontent.com/11625419/79631668-3b6c7180-819e-11ea-8fb7-c8a7fc948c11.png)

**To Reproduce**
1. Load some books - square covers are a good example
2. Use CaliBlur theme
3. Observe squashed cover images

**Expected behavior**
Book covers at full width but height appropriate to the aspect ratio:
![same book cover but square as desired](https://user-images.githubusercontent.com/11625419/79631810-92724680-819e-11ea-95de-f913bdd961fe.png)

**Screenshots**
See above.

**Environment (please complete the following information):**
- Docker, LinuxServer
- Firefox on Windows (latest)

**Fix**
This CSS fixes it, at least on a desktop view:
```
.book:not(body) {
  height: 295px;
  display: flex;
  justify-content: end;
  flex-direction: column;
}

.container-fluid .book .cover {
    height: auto;`
}
```

Basically this sets div.cover to auto height (not 275px) and sets the parent (div.book) to 295px to compensate. It then uses the three flex properties to align div.cover and div.meta to the bottom of div.book - so that books are aligned at the bottom:
![row of books of various heights, aligned on their bottom edge](https://user-images.githubusercontent.com/11625419/79631867-1c221400-819f-11ea-8609-9a3398e2953e.png)

This also maintains hover border and shading alignment with the cover:
![book cover with aligned hover fade and yellow border](https://user-images.githubusercontent.com/11625419/79631924-74f1ac80-819f-11ea-8895-06acf07bae83.png)

**Comment by OzzieIsaacs** [comment]
@hexeth could you have a look at this one, can we merge the suggestion without any sideffects?

**Comment by hexeth** [comment]
This is a little more complicated than the above code solves for. @leram84 did the CSS. I know a little, but it's definitely not my wheelhouse.

Not all elements are covered by above code:
![image](https://user-images.githubusercontent.com/7627137/79671901-98f2c500-8182-11ea-8089-0fca43727902.png)

Hover icon is no longer centered:
![image](https://user-images.githubusercontent.com/7627137/79672029-29c9a080-8183-11ea-9bde-97063ac3bf21.png)

**Comment by lyonzy** [comment]
That's weird, it looks like yours has kept div.book width at 225px - mine is overriden somewhere else at 180px:
![css screenshot](https://user-images.githubusercontent.com/11625419/79680193-4f70ab80-8250-11ea-9ea8-e7fe25e726c5.png)

Maybe @leram84 might know more about how this all works (maybe it's in a media query?).

Also I tested it in Chromium and it seems like it should actually be `justify-content: flex-end;` instead of just `end`.

**Comment by hexeth** [comment]
I noticed that removing `display: flex` made it all line up but I am missing any square book covers to see what it does to them

**Comment by hexeth** [comment]
Okay, think I see what's going on. The issue was that I was replacing the CSS with what you put above, but in reality it just needs to be added to the existing spots.

In caliblur.css 
```.container-fluid .book .cover {
    width: auto;
    display: inline-block;
    height: Auto;
}```

```.book:not(body) {
    width: 180px!important;
    max-width: 180px!important;
    min-width: 180px!important;
    height: 295px;
    display: flex;
    justify-content: end;
    flex-direction: column;
}```

Results in:
![image](https://user-images.githubusercontent.com/7627137/79697806-afa82300-8239-11ea-9cd1-b008f6a73083.png)

Not sure why yours line up

**Comment by hexeth** [comment]
It looks like

```
.book:not(body) {
    width: 180px!important;
    max-width: 180px!important;
    min-width: 180px!important;
    height: 295px;
    display: flex;
    justify-content: end;
    flex-direction: column;
    justify-content: flex-end
}
```

Solves this in Chrome for me

**Comment by Knepherbird** [comment]
"end", "start", etc., were originally developed for css grid contexts, and while there is an intention to replace flex-end, flex-start, etc., only FF and Safari have implemented support for them in flexbox context, so it's advised to stick with the flex-specific properties at this time. (See: https://caniuse.com/#feat=mdn-css_properties_justify-content_flex_context_start_end)

**Comment by hexeth** [comment]
Using `flex-end `instead of `end` does fix the problem in a way, but creates another strange one in Chrome:

![squished]( https://i.imgur.com/bptogPb.gif )

**Comment by Knepherbird** [comment]
I haven't dug through the whole stylesheet, but based on what's been posted here and some tinkering in dev inspector, I think these modified rules appear to provide a solution, including the Chrome issue just above. These are for desktop, of course, so may need modification for responsive breakpoints. As a side note, I'd love to see the bootstrap grid replaced with a modern css grid / flexbox solution.

```
#books>.cover>a {
    display: inline-block;
    width: auto;
    height: auto;
}

.book:not(body) {
    width: 180px!important;
    max-width: 180px!important;
    min-width: 180px!important;
    height: 295px;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
}

.container-fluid .book .cover {
    width: auto;
    display: inline-block;
    align-self: center;
    height: auto;
}

#books .cover img {
    width: auto !important;
    height: auto;
    object-fit: contain;
    max-height: 225px;
    max-width: 150px;
}
```

**Comment by chrisdma** [comment]
@OzzieIsaacs @hexeth Is there any reason why this hasn't been merged? I am using the Caliblur theme and have been experiencing this issue for quite a while and would love for it to be fixed.

**Comment by IceflowRE** [comment]
The latest suggested fix is not entirely working, it is still overlapping other elements when having a long title or series name

**Comment by IceflowRE** [comment]
This is a quick fix:
```css
.cover > a > span > img {
    object-fit: contain;
}
```

This wont scale the hover effect, but i like having the hover effect bigger anyway, makes it easier to click the covers.
